errors.go

```diff
@@ -92,11 +92,16 @@ func checkForErrors(res *http.Response) error {
 		res.Body.Close()
 	}()
 
+	out, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		return unknownServerError(res.StatusCode, err.Error())
+	}
+
 	switch res.StatusCode {
 	case 400:
 		bodyMap := make(map[string]interface{})
-		if err := json.NewDecoder(res.Body).Decode(&bodyMap); err != nil {
-			return unknownError(res.StatusCode, err)
+		if err := json.Unmarshal(out, &bodyMap); err != nil {
+			return unknownServerError(res.StatusCode, err.Error())
```

Comment:
I wanted to explicitly show the error that occurred when unmarshalling the JSON response body, but perhaps it's better to show the response that is causing `Unmarshal` to choke. What do you mean by

> this may not always return the full raw response body.

? :)